### PR TITLE
Updating Developer Guide README.md to include Hybrid search specific documentation

### PR DESCRIPTION
**Pull request by adityabharadwaj198**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
* docs update

* **What is the current behavior?** (You can also link to an open issue here)
* Currently the developer guide readme doesn't contain steps to create the Vespa custom searcher's .jar file, which would lead to a contributor encountering issues when following the developer guide

* **What is the new behavior (if this is a feature change)?**
* Updated readme file contains the steps to create the .jar file.  

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
* No

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
* Unit tests aren't required for the PR 

* **Related Python client changes** (link commit/PR here)
* None

* **Related documentation changes** (link commit/PR here)
* None

* **Other information**:

* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: src/marqo/README.md [hunk 1]
```diff
@@ -35,14 +35,56 @@ docker run --detach --name vespa --hostname vespa-tutorial \
 ```bash
 (cd scripts/vespa_local && zip -r - * | curl --header "Content-Type:application/zip" --data-binary @- http://localhost:19071/application/v2/tenant/default/prepareandactivate)
 ```
-
 You can verify that Vespa has been set up correctly by visiting `http://localhost:8080` in your browser.
 
+5. We've written custom logic to handle to reranking of all the results that Vespa outputs. We call this Hybrid search. Hybrid Search combines both lexical and tensor search to provide the best relevance possible. Hybrid search provides flexibility to get the best results of both types of search and fine tune which type to lean into. It's a great place to start when you want the precision of lexical search but the semantic understanding of tensor search.
```
Comment (farshidz) [inline src/marqo/README.md]: No need to refer to hybrid search here

File: src/marqo/README.md [hunk 2]
```diff
@@ -35,14 +35,56 @@ docker run --detach --name vespa --hostname vespa-tutorial \
...
+The logic is written in Java so you need to create a jar file 
+
+5a. Install Java Development Kit: 
+You can install the Java Development Kit (JDK) by following the instructions [here](https://openjdk.org/install/). 
+You will need to set Java to Path, you can do this by 
+```zsh
```
Comment (farshidz) [inline src/marqo/README.md]: Use bash to be consistent

File: src/marqo/README.md [hunk 3]
```diff
@@ -35,14 +35,56 @@ docker run --detach --name vespa --hostname vespa-tutorial \
...
+or 
+```zsh
+vi ~/.zprofile
+```
+In the .zshrc/.zprofile file, paste the following contents: 
+```
+JAVA_HOME="/Library/Java/JavaVirtualMachines/jdk-22.0.2.jdk/Contents/Home"
+PATH="${JAVA_HOME}/bin:${PATH}"
+```
+Remember to replace the "/Library/Java/JavaVirtualMachines/jdk-22.0.2.jdk/Contents/Home" with the path to your JDK.
```
Comment (farshidz) [inline src/marqo/README.md]: You could provide links but no need to cover how to install java etc in detail here

**Review by farshidz — CHANGES_REQUESTED** [review]

File: src/marqo/README.md (hunk 1 above)
Comment (adityabharadwaj198) [reply src/marqo/README.md]: removed hybrid search usage

File: src/marqo/README.md (hunk 3 above)
Comment (adityabharadwaj198) [reply src/marqo/README.md]: added links to install java/maven & removed details on installing

File: src/marqo/README.md (hunk 2 above)
Comment (adityabharadwaj198) [reply src/marqo/README.md]: changed to bash everywhere

File: src/marqo/README.md [hunk 4]
```diff
@@ -35,14 +35,35 @@ docker run --detach --name vespa --hostname vespa-tutorial \
...
+java -version
+```
+5b. Install Maven 
+You can install Maven by following the instructions [here](https://maven.apache.org/install.html).
+Similar to setting Java, you need to add Maven to the PATH Environment Variable.
+Verify Maven installation by running the following command: 
+```bash
+mvn -version
+```
+Post this you need to create a jar file, cd into the vespa directory in your local marqo repository, and run
```
Comment (farshidz) [inline src/marqo/README.md]: Make this step 6

File: src/marqo/README.md [hunk 5]
```diff
@@ -35,14 +35,35 @@ docker run --detach --name vespa --hostname vespa-tutorial \
...
+```
+5b. Install Maven 
+You can install Maven by following the instructions [here](https://maven.apache.org/install.html).
+Similar to setting Java, you need to add Maven to the PATH Environment Variable.
+Verify Maven installation by running the following command: 
+```bash
+mvn -version
+```
+Post this you need to create a jar file, cd into the vespa directory in your local marqo repository, and run 
+```bash
```
Comment (farshidz) [inline src/marqo/README.md:61]: `cd vespa`?

File: src/marqo/README.md (hunk 4 above)
Comment (farshidz) [inline src/marqo/README.md]: The wording is confusing. How do I create a jar file?

File: src/marqo/README.md [hunk 6]
```diff
@@ -35,14 +35,35 @@ docker run --detach --name vespa --hostname vespa-tutorial \
...
+mvn clean package
+```
+Post running this command, you will see that a target folder gets created in the vespa directory, which contains a jar file called marqo-custom-searchers-deploy.jar. This jar file is used to deploy the custom searchers to Vespa.
+
 ### Option A. Run the Marqo application locally (outside of docker) through IDE
 Now you can run Marqo locally through your IDE (e.g. PyCharm) by following the steps below.
 
-5. Open the Marqo project in your IDE (e.g. PyCharm) and go to the file `src/marqo/tensor_search/api.py`
-6. Set up your [debug configuration](https://www.jetbrains.com/help/pycharm/creating-run-debug-configuration-for-tests.html)
+6. Open the Marqo project in your IDE (e.g. PyCharm) and go to the file `src/marqo/tensor_search/api.py`
```
Comment (farshidz) [inline src/marqo/README.md]: (e.g. PyCharm) once is enough. After that it's redundant

File: src/marqo/README.md [hunk 7]
```diff
@@ -35,14 +35,35 @@ docker run --detach --name vespa --hostname vespa-tutorial \
...
+mvn -version
+```
+Post this you need to create a jar file, cd into the vespa directory in your local marqo repository, and run 
+```bash
+mvn clean package
+```
+Post running this command, you will see that a target folder gets created in the vespa directory, which contains a jar file called marqo-custom-searchers-deploy.jar. This jar file is used to deploy the custom searchers to Vespa.
+
 ### Option A. Run the Marqo application locally (outside of docker) through IDE
 Now you can run Marqo locally through your IDE (e.g. PyCharm) by following the steps below.
```
Comment (farshidz) [inline src/marqo/README.md]: e.g., (missing comma)

File: src/marqo/README.md [hunk 8]
```diff
@@ -52,14 +73,14 @@ VESPA_CONFIG_URL=http://localhost:19071
 VESPA_DOCUMENT_URL=http://localhost:8080
 VESPA_QUERY_URL=http://localhost:8080
 ```
-7. Now you can Debug this file directly from your IDE (e.g. PyCharm) to start Marqo locally.
-8. Set breakpoints in the project for better debugging experience.
+8. Now you can Debug this file directly from your IDE (e.g. PyCharm) to start Marqo locally.
```
Comment (farshidz) [inline src/marqo/README.md]: debug (lower case)

**Review by farshidz — CHANGES_REQUESTED** [review]

File: src/marqo/README.md (hunk 4 above)
Comment (adityabharadwaj198) [reply src/marqo/README.md]: done

Comment (adityabharadwaj198) [reply src/marqo/README.md]: changed

File: src/marqo/README.md (hunk 5 above)
Comment (adityabharadwaj198) [reply src/marqo/README.md:61]: added cd vespa in the line above

**Review by farshidz — APPROVED** [review]

**Pull request merged**